src/main/java/seedu/address/logic/parser/editcommandparser/EditCommandAllocator.java

```diff
@@ -16,9 +16,15 @@
 
     @Override
     public EditCommand allocate(String userInput) throws ParseException {
-
-        String entity = AlfredParserUtil.getEntityFromCommand(userInput, EditCommand.MESSAGE_USAGE);
-        String args = AlfredParserUtil.getArgumentsFromCommand(userInput, EditCommand.MESSAGE_USAGE);
+        String entity;
+        String args;
+
+        try {
+            entity = AlfredParserUtil.getSpecifierFromCommand(userInput);
+            args = AlfredParserUtil.getArgumentsFromCommand(userInput);
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(pe.getMessage(), EditCommand.MESSAGE_USAGE));
```

Comment:
`MESSAGE_INVALID_COMMAND_FORMAT` instead?

Reply:
Good point. Although pe.getMessage() would return `MESSAGE_INVALID_COMMAND_FORMAT` , I replaced it to make it more understandable.